### ISSUE TITLE
Problem with continuing_subword_prefix?

### ISSUE THREAD

**Issue by dimkart** [huggingface/tokenizers#287]
I train my own BPE model using a `continuing_subword_prefix` equal to `##`. However, when I try to  load the model by using `models.BPE.from_files` method, I'm getting the following error:

```python
Traceback (most recent call last):
  File "bpe-tokenize.py", line 68, in <module>
    continuing_subword_prefix=args.subword_prefix
Exception: Token `##e##x` out of vocabulary
```

The error happens when invoking `from_files`. The merges files contains the entry ##e ##x, which has been merged to  new subtoken ##ex; this subtoken is included in the vocabulary. However, the `from_files` method is wrongly looking for a `##e##x` token, which of course is not there. Is this a bug or am I doing something wrong? Thanks in advance.

**Comment by dimkart** [comment]
Any comments on this please? Is it a bug of the code?

**Comment by dimkart** [comment]
That seems like a genuine bug. Replacing the computation of `new_token` in method `convert_merges_to_hashmap`  (`tokenizers/src/models/bpe/model.rs`) with the following (I have just hard-coded the prefix I am using just for test) fixes the problem. 

```rust
let new_token;
if &parts[1][..2] == "##" {
      new_token = format!("{}{}", parts[0], &parts[1][2..]);
} else {
      new_token = format!("{}{}", parts[0], parts[1]);
}
```

Can someone verify this?

**Comment by n1t0** [comment]
Fixed by faaef548791447db9a55b4e403da92aa59159fab
